src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+    }",
+                GetCA3075XPathDocumentCSharpResultAt(12, 37, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(8, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInCatchBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+   using System;
+   using System.Xml.XPath;
+
+    class TestClass
+    {
+        private void TestMethod()
+        {
+            try { }
+            catch (Exception)
+            {
+                var xml = """";
+                XPathDocument doc = new XPathDocument(xml);
+            }
+            finally { }
+        }
+    }",
+                GetCA3075XPathDocumentCSharpResultAt(13, 37, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInFinallyBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private void TestMethod()
+    {
+        try { }
+        catch (Exception) { throw; }
+        finally
+        {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        }
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(14, 33, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(11, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInAsyncAwaitShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Threading.Tasks;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private async Task TestMethod()
+    {
+        await Task.Run(() => {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        });
+    }
+
+    private async void TestMethod2()
+    {
+        await TestMethod();
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "Run")
```

Comment:
And I think the point of this test was to show that when the diagnostic occurred in an anonymous method, you stepped up the syntax tree until you found a named method to report. Again, if you don't mention the method name, you don't need this test.